src/integrations/watchers/addon-update-watcher.php

```diff
@@ -0,0 +1,166 @@
+		}
+
+		return $old_html;
+	}
+
+	/**
+	 * Enables premium auto updates when free are enabled and the other way around.
+	 *
+	 * @param array $old_value The old value of the `auto_update_plugins` option.
+	 * @param array $new_value The new value of the `auto_update_plugins` option.
+	 *
+	 * @return void
+	 */
+	public function toggle_auto_updates_for_add_ons_on_free_auto_update_toggle( $old_value, $new_value ) {
+		if ( ! \is_array( $old_value ) || ! \is_array( $new_value ) ) {
+			return;
+		}
+
+		if ( $this->auto_updates_for_free_have_just_been_enabled( $old_value, $new_value ) ) {
+			$this->enable_auto_updates_for_addons( $new_value );
+		}
+		elseif ( $this->auto_updates_for_free_have_just_been_disabled( $old_value, $new_value ) ) {
+			$this->disable_auto_updates_for_addons( $new_value );
+		}
+	}
+
+	/**
+	 * Checks whether auto updates for free have just been enabled.
+	 *
+	 * @param array $old_value The old value of the `auto_update_plugins` option.
+	 * @param array $new_value The new value of the `auto_update_plugins` option.
+	 *
+	 * @return bool Whether auto updates for free have just been enabled.
+	 */
+	protected function auto_updates_for_free_have_just_been_enabled( $old_value, $new_value ) {
+		return $this->auto_updates_are_enabled_for_free( $new_value ) && ! $this->auto_updates_are_enabled_for_free( $old_value );
+	}
+
+	/**
+	 * Checks whether auto updates for free have just been disabled.
+	 *
+	 * @param array $old_value The old value of the `auto_update_plugins` option.
+	 * @param array $new_value The new value of the `auto_update_plugins` option.
+	 *
+	 * @return bool Whether auto updates for free have just been disabled.
+	 */
+	protected function auto_updates_for_free_have_just_been_disabled( $old_value, $new_value ) {
+		return $this->auto_updates_are_enabled_for_free( $old_value ) && ! $this->auto_updates_are_enabled_for_free( $new_value );
+	}
+
+	/**
+	 * Enables auto-updates for all addons.
+	 *
+	 * @param string[] $auto_updated_plugins The current list of auto-updated plugins.
+	 */
+	protected function enable_auto_updates_for_addons( $auto_updated_plugins ) {
+		\update_option( 'auto_update_plugins', \array_merge( $auto_updated_plugins, self::ADD_ONS ) );
+	}
+
+	/**
+	 * Disables auto-updates for all addons.
+	 *
+	 * @param string[] $auto_updated_plugins The current list of auto-updated plugins.
+	 */
+	protected function disable_auto_updates_for_addons( $auto_updated_plugins ) {
+		\update_option( 'auto_update_plugins', \array_diff( $auto_updated_plugins, self::ADD_ONS ) );
+	}
+
+	/**
+	 * Checks whether auto updates for Yoast SEO are enabled.
+	 *
+	 * @param array $auto_updated_plugins The array of auto updated plugins.
+	 *
+	 * @return bool Whether auto updates for Yoast SEO are enabled.
+	 */
+	protected function auto_updates_are_enabled_for_free( $auto_updated_plugins ) {
```

Comment:
Have you considered passing the plugin/addon id? That will make this method reusable for line https://github.com/Yoast/wordpress-seo/pull/16694/files#diff-a6503fe42ced0c9408d6e40042140f842657eee9cc467c7bda3f136aa42d0c08R76

With that in mind you can rename this method to something that ask a question, just because it return a boolean. And you can remove the `free` part

I would think of: `are_auto_updates_enabled` where you pass the plugin id and a list of plugins :)